### DISCUSSION TITLE
Anyway to launch Epic and Steam games directly from moonlight?

### DISCUSSION THREAD

**Discussion by vhenzor** [LizardByte/community#65] [Sunshine] [answered]
First I want to thanks the developers for this host program, I recently bought my first amd gpu and was not perfectly satisfy with parsec coming from GFE/moonlight, finding sunshine was a life changer.

I can't figure out how to make my Epic and Steam games to launch directly from moonlight applications. I tried to add the url found in the desktop shortcut in the "Command" line of sunshine WebUI but always got a "Connection Error : Failed to start (error -1)".

Any idea?

CPU : Ryzen 5600
GPU : RX 6700 XT
OS : Windows 11
Sunshine v0.14.0 as service

**Comment by DaddyW123** [comment]
I came here looking for the same answer.  Glad I'm not the only one trying to do this, but sad there wasn't already a reply.  Adding this comment to hopefully gain traction on it.

**Reply by vhenzor** [reply]
Actually I ended creating a batch script who start the url shortcut for each of my Epic and Steam games and linked theirs .bat files as a detached commands in sunshine.

1- You first need to place a copy of all your Epic and Steam games shortcut in a folder, it will make things easier.

2- Then create a .bat file following this script :
```
@ECHO OFF
cd /d C:\Games\Shortcut 
start "" "Rocket League.lnk"
```

C:\Games\Shortcut = the path of your shorcut folder
"Rocket League.lnk" = the name of your shortcut

3- Save and repeat for all your games.

4- Add the .bat files in sunshine as a detached commands

**Reply by ReenigneArcher** [reply]
You don't need a bat files for this... your detached command could just be

`cd /d C:\Games\Shortcut && start "" "Rocket League.lnk"`

**Comment by ReenigneArcher** [comment]
URIs need to be a disconnected command.

You may be able to use a "command" but definitely not if you're launching the URI. Here's a list of steam command line args. https://developer.valvesoftware.com/wiki/Command_Line_Options#Command-Line_Parameters_4

Those docs indicate that `steam -applaunch <appID>` may work as you're expecting... it still may require to be a disconnected command though.

I don't know what the command line args for Epic.

**Comment by vhenzor** [comment, marked as answer]
Actually I ended creating a batch script who start the url shortcut for each of my Epic and Steam games and linked theirs .bat files as a detached commands in sunshine.

1- You first need to place a copy of all your Epic and Steam games shortcut in a folder, it will make things easier.

2- Then create a .bat file following this script :
```
@ECHO OFF
cd /d C:\Games\Shortcut 
start "" "Rocket League.lnk"
```

C:\Games\Shortcut = the path of your shorcut folder
"Rocket League.lnk" = the name of your shortcut

3- Save and repeat for all your games.

4- Add the .bat files in sunshine as a detached commands

**Comment by AnicetusCer** [comment]
The answers above will probably all work but there is an even simpler way, use the steam URI scheme command (note; not the command line command. )
so instead of using "steam -applaunch 1234567", which will not work as a direct command in sunshine, use this uri version of the command instead:
 
steam://rungameid/1234567

Example
App id's can be found simpily by browsing to your desired game in in the steam store in a browser and looking at the address bar , you should see a number for the game in the address.
something like: store.steampowered.com/app/1234567/TheGame/

Then just add the id into the sunshine command field.
steam://rungameid/1234567
